aws-android-sdk-iot/src/test/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManagerTest.java

```diff
@@ -273,6 +273,31 @@ public void testConnectWebSocket() throws Exception {
         assertEquals(MqttManagerConnectionState.Connected, testClient.getConnectionState());
     }
 
+    @Test
+    public void testWebsocketConnectionWithUsernamePassword() throws Exception {
+        // Given: the MqttClient and an instance of AWSIotMqttManager.
+        MockMqttClient mockClient = new MockMqttClient();
+        AWSIotMqttManager awsIotMqttManager = new AWSIotMqttManager("test-client",
+                Region.getRegion(Regions.US_EAST_1), TEST_ENDPOINT_PREFIX);
+        awsIotMqttManager.setMqttClient(mockClient);
+        TestClientStatusCallback csb = new TestClientStatusCallback();
+
+        // When: the SDK tries to connect to IoT.
+        awsIotMqttManager.connect("user", "password", csb);
```

Comment:
Instead of waiting an arbitrary amount of time, wait for the even of interest -- the callback occurring. Perhaps, something like:
```
CountDownLatch latch = new CountDownLatch(1);
awsIotMqttManager.connect("user", "password", new TestClientStatusCallback() {
    @Override
    public void onSomeStatusEventCallback(StatusInputType type) {
        super.onSomeStatusEventCallback(type); // Same behavior so far...
        latch.countDown();
    }
});
latch.await(500, TimeUnit.MILLISECONDS);
```
^ In this case, the arbitrary value is the _upper bound_ for the result. But, most of the time, the result will be rendered _before_ 500ms has elapsed.